### DISCUSSION TITLE
Supabase Edge Function - SSL Certiface problem

### DISCUSSION THREAD

**Discussion by adrianarciszewski95** [supabase/supabase#22833] [Questions] [answered]
Hi,

I am currently utilizing a self-hosted version of Supabase on our company’s infrastructure, where we leverage webhooks and edge functions to manage various events in our database.

We are facing a challenge with one of our edge functions. We need to make a GET request to an SMS gateway API, which requires authentication using an SSL certificate stored in a .pem file. Unfortunately, we found that Deno does not yet support any libraries or tools that allow for custom certificates in requests. We have tried using Deno.HttpClient, but it is not stable so we cannot use within Supabase edge functions. We also could not find an equivalent solution using built-in Node APIs or modules published to npm.

Is there any clear solution to solve this problem in Supabase?

**Comment by laktek** [comment, marked as answer]
@adrianarciszewski95 You can add the custom SSL certificate to your system (https://unix.stackexchange.com/questions/90450/adding-a-self-signed-certificate-to-the-trusted-list).

Then when running the edge-runtime, set the environment variable `DENO_TLS_CA_STORE` to `mozilla,system`. That should allow edge-runtime to pick up the custom cert from the system.
